indexing/src/main/scala/org/allenai/common/indexing/BuildCorpusIndex.scala

```diff
@@ -238,21 +238,58 @@ class BuildCorpusIndex(config: Config) extends Logging {
 
   /** Index a single segment into elasticsearch.
     * @param segment to be indexed
+    * @param documentFormat also describes the format of the segment
     * @param source name of source for reference
     * @param segmentIndex index of segment in file (for deduplication)
     * @param bulkProcessor to communicate with the elasticsearch instance
     */
   def addSegmentToIndex(
     segment: String,
+    documentFormat: String,
     source: String,
     segmentIndex: Int,
     bulkProcessor: BulkProcessor
   ): Unit = {
-    val request = new IndexRequest(indexName, indexType).source(jsonBuilder().startObject()
-      .field("text", segment.trim)
-      .field("source", source + "_" + segmentIndex.toString)
-      .endObject())
-    bulkProcessor.add(request)
+    // Helper Function
+    def breakQAline(line: String): Option[(String, String)] = {
+      line.split(""":\|:""") match {
```

Comment:
You can skip the double-trim if you want to: `line.split(...).map(_.trim) match { ... }`